### DISCUSSION TITLE
how to upload file from pc to github  by code

### DISCUSSION THREAD

**Discussion by SonwTungf** [community/community#128394] [New to GitHub]
how to upload file from pc to github  by code

**Comment by IshworTM** [comment]
Hi, @SonwTungf, you can simply do that by opening up your vs-code and then opening the terminal you can do that by pressing ``Ctrl + ` `` and then browse to the directory containing your files that you want to push to github. After doing that open up your GitHub then click on **Your Repositories** then click on the **New** icon:
![image](https://github.com/community/community/assets/98617144/fb572894-cab8-4660-8f87-44804866bc5f)
After making a new repository copy the **SSH** link

![image](https://github.com/community/community/assets/98617144/953faf1d-1d77-4cc9-ad56-bb5e7a4075fd)

Then you can go back to your vs-code terminal and then execute the following commands:
`git init` - To initialise an empty git repository so that you can connect it to a remote one.
`git remote add origin git@github.com:YourRepository's Link.git` - To connect your local empty repository to the remote repository.
`git branch -M main` - To add and move to a **main** branch, it will be **master** by default. 
`git add <filename>` - To add a specific file so that you can push it to your GitHub repository.
`git add .` - To add all the files on your current working directory to your GitHub repository.
`git commit -m "your commit message"` - To commit your changes so that it can be reflected when you push the changes to GitHub.
`git push -u origin main` - To finally push your changes to your GitHub repository.

If you prefer, you can watch this YouTube [tutorial](https://www.youtube.com/watch?v=rLkz5UzAX_8&t=99s&pp=ygUWaXNod29yIGdpdGh1YiB0dXRvcmlhbA%3D%3D) as well.

After following these instructions, you can check your GitHub repository, all the files you added should be shown there. Hope this helps. :)

**Comment by erroradib** [comment]
response #2

**Comment by rockett-m** [comment]
inside the repo on your computer
```console
git status
git add filename
git commit -m "commit message"
git push
```
